packages/rum/src/userActionCollection.ts

```diff
@@ -71,18 +73,33 @@ export function startUserActionCollection(lifeCycle: LifeCycle) {
 
 function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string) {
   if (pendingAutoUserAction) {
-    // Discard any new user action if another one is already occurring.
+    // Ignore any new user action if another one is already occurring.
     return
   }
 
   const id = generateUUID()
   const startTime = performance.now()
 
+  lifeCycle.notify(LifeCycleEventType.ACTION_CREATED)
+
   const { eventCounts, stop: stopEventCountsTracking } = trackEventCounts(lifeCycle)
 
   const { stop: stopWaitIdlePageActivity } = waitIdlePageActivity(lifeCycle, (hadActivity, endTime) => {
+    if (!pendingAutoUserAction) {
+      return
+    }
```

Comment:
not super happy with that, but I have some ideas that would be easier to implement once getUserActionReference will be replaced

Reply:
I would be even more careful by making sure the `pendingAutoUserAction` is the same as the one created earlier by `newUserAction`:

When initializing pendingAutoUserAction, do something like this:
```js
const autoUserAction = {...}
pendingAutoUserAction = autoUserAction
```

and then you can replace the condition to:

```suggestion
    if (pendingAutoUserAction === autoUserAction) {
      return
    }
```

Reply:
yes, I have experimented with something like that but I would prefer to rework the design to remove the need for the global variable. It should be easier after coming PRs so I preferred to keep it simple for now.
Do you see new risks introduce by these changes?

Reply:
It should be fine!